sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -111,12 +111,11 @@ case class CreateViewCommand(
 
     // When creating a permanent view, not allowed to reference temporary objects.
     // This should be called after `qe.assertAnalyzed()` (i.e., `child` can be resolved)
-    verifyTemporaryObjectsNotExists(catalog, isTemporary, name, child)
+    verifyTemporaryObjectsNotExists(catalog, isTemporary, name, analyzedPlan)
```

Comment:
since `child` is already resolved, we don't need to call `sparkSession.sessionState.executePlan(child)` and re-analyze it now.

Reply:
`CacheTableAsSelectExec.query` may not be analyzed? (other places seem fine)

Reply:
Just analyze it in `CacheTableAsSelectExec` and remove analyzing here?

Reply:
Ok let's fix it in followup then. We probably should also fix `CacheTableAsSelect` and resolve the `query` in the analyzer.

Reply:
@cloud-fan if we remove `sparkSession.sessionState.executePlan(child)` and use `child` directly, do you think there would be an issue similar to `CacheTable` since `child` will be an optimized plan, not an analyzed plan?